packages/ketcher-core/src/application/editor/modes/SequenceMode.ts

```diff
@@ -821,6 +821,48 @@ export class SequenceMode extends BaseMode {
     return true;
   }
 
+  private isR1Free(firstNode: SubChainNode): boolean {
+    return firstNode?.firstMonomerInNode?.attachmentPointsToBonds?.R1 === null;
+  }
+
+  private isR2Free(lastNode: SubChainNode): boolean {
+    return lastNode?.lastMonomerInNode?.attachmentPointsToBonds.R2 === null;
+  }
+
+  private areR1R2Free(
+    firstNode: SubChainNode,
+    lastNode: SubChainNode,
+  ): boolean {
+    return this.isR1Free(firstNode) && this.isR2Free(lastNode);
+  }
+
+  isPasteAvailable(drawingEntitiesManager: DrawingEntitiesManager) {
+    if (!this.isEditMode) {
+      return true;
+    }
+    const chainsCollection = ChainsCollection.fromMonomers([
+      ...drawingEntitiesManager.monomers.values(),
+    ]);
+    const currentNode = SequenceRenderer.currentEdittingNode;
+    const previousNodeInSameChain = SequenceRenderer.previousNodeInSameChain;
+    const lastNodeOfNewFragment = chainsCollection.lastNode;
+    const firstNodeOfNewFragment = chainsCollection.firstNode;
+    const isPasteInEnd =
+      currentNode instanceof EmptySequenceNode || !currentNode;
+    const isPasteInStart = !previousNodeInSameChain;
+    if (isPasteInEnd && !previousNodeInSameChain) return true;
+    if (isPasteInEnd) {
+      return (
+        this.isR1Free(firstNodeOfNewFragment) &&
+        this.isR2Free(previousNodeInSameChain)
+      );
+    }
+    if (isPasteInStart) {
+      return this.isR2Free(lastNodeOfNewFragment) && this.isR1Free(currentNode);
```

Comment:
If you need to check both R1 and R2 when pasting into start or end of sequence, you may probably use `areR1R2Free` here rather than separate functions